partner_contact_lang/models/res_partner.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# © 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.multi
+    def write(self, vals):
+        if 'lang' in vals and vals['lang']:
```

Comment:
Hmm I understand what you mean. However, in this case is better `vals.get("lang")`, because otherwise, if you are editing a parent partner and unsetting its lang, then you would search for all children that have no lang and unset it (which is redundant).